### PR TITLE
Put layout options into the menu

### PR DESCRIPTION
**Pull request by jonbell-lot23**
**Review note**

Brian, you're the reviewer so as always I'm looking for ways to make the code better. But please note that we may have design feedback coming through, so this PR may stay open for a bit while we discuss any concerns.

**Background**

This has been a long-term problem with our product. People expect to be able to pin their panes to the side or bottom, but most people don't discover the functionality. This finally makes the feature discoverable and addresses https://linear.app/replay/issue/DES-743/[explore]-put-layout-options-into-our-global-menu. ([Figma](https://www.figma.com/file/mAotNZRv6M5X0zHIFKPzFY/Replay-Design-Doc?type=design&node-id=18487-160698&mode=design&t=cdbbM6HKVfap7sWT-4))

[Loom of the feature in action](https://www.loom.com/share/dad97764c1bc435cae9137657fae97d9)

Screenshot:
image

### CODE REVIEW

File: src/ui/components/Header/UserOptions.tsx [hunk 1]
```diff
@@ -34,10 +43,21 @@ function UserOptions({ setModal, noBrowserItem }: UserOptionsProps) {
...
+    dispatch(setToolboxLayout("ide"));
+  };
+
+  const onLayoutDockLeftClick = () => {
+    dispatch(setToolboxLayout("left"));
+  };
+
+  const onLayoutDockBottomClick = () => {
+    dispatch(setToolboxLayout("bottom"));
+  };
```
Comment (jasonLaster) [inline src/ui/components/Header/UserOptions.tsx:59]: nit: I'd do `const onLayoutChange = (orientation) => dispatch(setToolboxLayout(orientation));` again pretty minor

Comment (jonbell-lot23) [reply src/ui/components/Header/UserOptions.tsx:59]: Thanks! Will look into that

**Comment by bvaughn** [comment]
> But please note that we may have design feedback coming through, so this PR may stay open for a bit while we discuss any concerns.

I'd prefer not to do multiple rounds of review. Tag me when the design is stable and I'll review.

**Comment by jonbell-lot23** [comment]
> > But please note that we may have design feedback coming through, so this PR may stay open for a bit while we discuss any concerns.
> 
> I'd prefer not to do multiple rounds of review. Tag me when the design is stable and I'll review.

@bvaughn I think we're good. Jason had a code suggestion and said shipit to the design itself.

File: src/ui/components/Header/UserOptions.tsx [hunk 2]
```diff
@@ -53,25 +65,58 @@ function UserOptions({ setModal, noBrowserItem }: UserOptionsProps) {
...
+            <span className="flex-1 text-left">Chat with us</span>
             <Icon filename="help" className="bg-iconColor" />
-            <span>Chat with us</span>
           </ExternalLink>
-          <button className="row group" onClick={onSettingsClick}>
-            <Icon filename="settings" className="bg-iconColor" />
-            <span>Settings</span>
-          </button>
-          {window.__IS_RECORD_REPLAY_RUNTIME__ || noBrowserItem ? null : (
-            <button className="row group" onClick={onLaunchClick}>
```
Comment (bvaughn) [inline src/ui/components/Header/UserOptions.tsx:69]: The `onLaunchClick` event handler should be deleted (above) since it's no longer being referenced.

**Comment by jonbell-lot23** [comment]
@bvaughn Adjusted based on feedback. Also added trackEvent.

**Review by bvaughn — APPROVED** [review]

File: src/ui/components/Header/UserOptions.tsx [hunk 3]
```diff
@@ -53,25 +61,58 @@ function UserOptions({ setModal, noBrowserItem }: UserOptionsProps) {
...
+                        toolboxLayout === "bottom" ? styles.selected : ""
+                      }`}
+                    />
+                  </div>
+                </div>
+              </div>
+
+              <div className={styles.divider}></div>
+            </>
+          )}
```
Comment (bvaughn) [inline src/ui/components/Header/UserOptions.tsx:99]: I think we also regressed for Node recordings with this change (`recordingCapabilities.supportsRepaintingGraphics`). I totally missed that.

**Comment by bvaughn** [comment]
I didn't really notice this either but the "Layout" row shouldn't be a button with a hover state:
Screen Shot 2023-08-29 at 1 05 52 PM

Because clicking it doesn't do anything.

**Comment by bvaughn** [comment]
I'll address the above via FE-1858

**Pull request merged**